filtered-resource-service/src/main/java/uk/gov/gchq/palisade/service/filteredresource/repository/exception/TokenErrorMessageController.java

```diff
@@ -0,0 +1,165 @@
+        return ActorFlow.ask(tokenEMCommand, TIMEOUT, SpawnWorker::new)
+                .map((WorkerResponse workerResponse) -> {
+
+                    //If there are TokenErrorMessageEntities retrieved from persistence for the token
+                    if (workerResponse instanceof SetAuditErrorMessages) {
+                        SetAuditErrorMessages setAuditErrorMessages = (SetAuditErrorMessages) workerResponse;
+                        return TokenErrorMessagePersistenceResponse.Builder.create()
+                                .withToken(setAuditErrorMessages.token)
+                                .withMessageEntities(setAuditErrorMessages.messageEntities);
+
+                        // If there was an issue retrieving the message from persistence then report an error
+                    } else if (workerResponse instanceof ReportError) {
+                        ReportError reportError = (ReportError) workerResponse;
+                        return TokenErrorMessagePersistenceResponse.Builder.create()
+                                .withToken(reportError.token)
+                                .withException(reportError.exception);
+                    } else {
+                        // This can only occur if the WorkerResponse subclasses are extended and not accounted for here
+                        // Entirely developer error, and provides a return from this if statement
+                        throw new NoSuchElementException(String.format(
+                                "Could not determine input class type %s as a subclass of %s",
+                                workerResponse.getClass(), WorkerResponse.class));
+                    }
+                });
+    }
+
+    /**
+     * Default (initial) behaviour for this actor to assume.
+     *
+     * @return a behaviour that accepts all {@link TokenErrorMessageCommand}s (this actor does not 'become' any other behaviour)
+     */
+    @Override
+    public Receive<TokenErrorMessageCommand> createReceive() {
+        return newReceiveBuilder()
+                .onMessage(SpawnWorker.class, this::onSpawnWorker)
+                .build();
+    }
+
+    /**
+     * The {@link SpawnWorker} command will spawn a {@link TokenErrorMessageWorker} and pass it the command's token, ({@link Pair#first()}).
+     * After the worker has been spawned, it is added to the getAllExceptions and told to look for AuditErrorMessages that link to a token.
+     * They are given a UUID to remain unique.
+     * The command's {@code replyTo} will later {@link ActorRef#tell} the AuditErrorMessages for that token.
+     *
+     * @param spawnWorker the {@link SpawnWorker} command to handle
+     * @return an unchanged {@link Behavior} (this actor does not 'become' any other behaviour)
+     */
+    private Behavior<TokenErrorMessageCommand> onSpawnWorker(final SpawnWorker spawnWorker) {
+        // Spawn a new worker, passing this (its parent) as a 'callback' for Deregister commands
```

Comment:
Not using deregisters
```suggestion
        // Spawn a new worker, passing it the persistenceLayer to query
```

Reply:
updated